### ISSUE TITLE
Remove "generated from" 

### ISSUE THREAD

**Issue by sandstrom** [ruby/setup-ruby#124]
Is it possible to remove the text "generated from actions/javascript-action"?

It's confusing and makes this look like a non-official repo.

**Comment by eregon** [comment]
Agreed it's confusing.
I didn't see a way in the settings, I'll try to ask GitHub Support.

**Comment by eregon** [comment]
I filed https://support.github.com/ticket/personal/0/938692 (but probably only I can see it).

**Comment by deivid-rodriguez** [comment]
I unforked repos in the past, github support can handle it :+1:.

**Comment by eregon** [comment]
Here is the reply, in short it's not possible to remove `generated from` currently:

> I understand that this is not an ideal response and apologize that I am unable to remove that label for you at this time.
> 
> The only possible workaround at this time is hacky at best. The steps to do that are as follows
> 
> change directory to a new location to not overwrite the existing one
> git clone --bare https://github.com/ruby/setup-ruby.git
> Rename the existing repository to indicate it is not in use like setup-ruby_do_not_use.
> Create a new repository with the same name setup-ruby
> cd repo.git
> git push --mirror https://github.com/ruby/setup-ruby.git
> This is basically a reworking of the Import a repository to import GitHub --> GitHub.
> 
> Again, I apologize that this is not easier and I have given this feedback to our product team as well, though I can't say if when this ability may be added.

Recreating the repository like that is not an option, we would lose all releases and probably links to the GitHub Actions Marketplace, etc.

So I guess the only approach until this become a setting would be to suggest a better way to show that `generated from` information to GitHub.
Right now it looks too similar to `forked from`, which is quite confusing.
IMHO `generated from` is far less useful for most people looking at it, it's only useful for people wanting to e.g. start a similar action, so it could be somewhere near `About` along with other repository information.
Other ideas?

I'll close this because I don't think I can do more about it.

**Comment by deivid-rodriguez** [comment]
Oh, I misunderstood things I thought this was a fork! So, yeah, definitely agree on it being confusing. Thanks for trying @eregon.

**Comment by eregon** [comment]
I'll send a ticket about moving that `generated from` label somewhere, reopening so I don't forget to track this.

**Comment by eregon** [comment]
I filed a suggestion to move the `generated from` label somewhere else: https://support.github.com/ticket/personal/0/939881
FWIW the request is copied below:

---

Currently the `generated from` label, for repos which were created from a template repository is shown very prominently:
![Screenshot from 2020-12-11 18-07-48.png](https://github.zendesk.com/attachments/token/YKpnnIVkdxEFbL4b5d5JnOjhJ/?name=Screenshot+from+2020-12-11+18-07-48.png)

This actually confused many people, at the very least 3 people already mentioned it on https://github.com/ruby/setup-ruby/issues/124 (I guess hundreds in practice got confused, but those 3 actually mentioned it there).
The issue is most people will confuse this label with the `forked from` label, and to them any text under the repo name shown so prominently means "non-official repo, must be a fork".
See how similar they look:
![Screenshot from 2020-12-12 14-41-26.png](https://github.zendesk.com/attachments/token/paPzXhYGg7Myr5p1eYslXlbE8/?name=Screenshot+from+2020-12-12+14-41-26.png)

So I would like to strongly suggest to move the `generated from` label somewhere else.
The fact it was created from a template repository is irrelevant for most people interacting with that repository.
In other words, it's only relevant for people who would like to create a similar repository, in this case e.g. another setup-something action.

I suggest to move it on the right bar next to other repository information, somewhere in here:
![Screenshot from 2020-12-12 14-44-17.png](https://github.zendesk.com/attachments/token/YtqUdjeG6XNIre4jYBreNG7UK/?name=Screenshot+from+2020-12-12+14-44-17.png)
Somewhere in the `About` or `Languages` section seem to fit well, or it could be its own section even.

Note that I already requested removing that `generated from` label, but I was told it's not possible in https://support.github.com/ticket/personal/0/938692.

**Comment by eregon** [comment]
I got a very quick reply, now it's in GitHub's hands:

> Thank you for sharing your thoughts, comments, and ideas with us! We have forwarded your ticket to the Product team who reads and evaluates all feedback, however we cannot guarantee a response to every submission.
>
> Please note that this means the Developer Support team will no longer be able to see changes to the ticket, so if you have a follow up that requires further support, please create a new ticket. Thank you again for your suggestion!

**Comment by sandstrom** [comment]
@thboop is this something you can help out with? 

Pinging since you've been involved in the sibling action 'actions/setup-ruby'.

**Comment by sandstrom** [comment]
How about renaming this repo to `setup-ruby-old` and create a new repo called `setup-ruby`, then populate that new repo with all the old git commits (i.e. same SHAs etc).
 
That way you'd get rid of the "generated from" text.

**Comment by hsbt** [comment]
We can remove the relationship of `actions/javascript-action` by GitHub Support. I think it's better than regenerate the new repo.

**Comment by sandstrom** [comment]
@hsbt Good idea! 💯

**Comment by eregon** [comment]
> We can remove the relationship of `actions/javascript-action` by GitHub Support. I think it's better than regenerate the new repo.

Feel free to ask it, but last time I asked it was not possible:
![Screenshot from 2021-12-01 15-28-08](https://user-images.githubusercontent.com/168854/144252362-56e1fe04-df7e-4748-b13b-583e8cb93162.png)

Recreating/renaming the repo is a strict no given the releases and association with the GitHub marketplace.

**Comment by eregon** [comment]
I think it's worth a shot to ask again, @hsbt could you do so?

**Comment by hsbt** [comment]
Sorry, I misunderstand it with "forked from". I don't know cut the repository network for "generated from".

**Comment by deivid-rodriguez** [comment]
You're not the only one @hsbt, it's so confusing. GitHub should redesign this marker.
